lib/logstash/filters/grok.rb

```diff
@@ -404,16 +408,34 @@ def add_patterns_from_files(paths, grok)
       end
     end # def add_patterns_from_files
 
-    private
     def add_patterns_from_inline_definition(pattern_definitions, grok)
       pattern_definitions.each do |name, pattern|
         next if pattern.nil?
         grok.add_pattern(name, pattern.chomp)
       end
     end
 
-    def close
-      @timeout_enforcer.stop!
-    end
+    class TimeoutError < RuntimeError; end
+
+    class GrokException < Exception
```

Comment:
Nitpick: I think this should be `GrokTimeoutException`, as the handling when caught is still very specific to timeouts.